### PR TITLE
add a secondary method for local socket check

### PR DESCRIPTION
**Pull request by birarda**
- added a secondary check for local IP information

### CODE REVIEW

**Comment by birarda** [comment]
I'll be testing this with @imgntn and @huffman - it attempts to work around a case where the guessed local IP could be wrong and local hole punching would fail.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 20ab108d51a3bad8469197a1cc2e57217df63a6a available at: https://gist.github.com/137de039bb9ac01a8859<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7270.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7270.dmg

**Comment by birarda** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 20ab108d51a3bad8469197a1cc2e57217df63a6a available at: https://gist.github.com/137de039bb9ac01a8859<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7270.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7270.dmg

**Comment by imgntn** [comment]
works great for me - win 10 -- with virtualbox host enabled, i cant connect to anything on master, with this PR i can connect!

**Comment by ZappoMan** [comment]
What bug is this fixing? What's the repro case?

**Comment by birarda** [comment]
Fixes https://app.asana.com/0/26225263936266/98191135940054. Repro case is specific to a network adapter config that @imgntn has on his machine.

**Comment by imgntn** [comment]
i'm using virtualbox -- it's not that unique of a conig, i don't think -- would probably happen for parallels or vmware too -- there's a virtual ethernet adapter that interface seemed previously to pick up before my actual ethernet adapter

**Comment by hifi-gustavo** [comment]
Build console output for commit: 909979bab9e31132461f2d92098fd5ca19b473c1 available at: https://gist.github.com/137de039bb9ac01a8859<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7270.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7270.dmg

**Comment by birarda** [comment]
Tested this with @huffman and it covered his case as well (with a loopback adapter for pcap installed)

File: libraries/networking/src/LimitedNodeList.cpp [hunk 1]
```diff
@@ -886,17 +886,65 @@ void LimitedNodeList::stopInitialSTUNUpdate(bool success) {
...
-    HifiSockAddr newSockAddr(getLocalAddress(), _nodeSocket.localPort());
-    if (newSockAddr != _localSockAddr) {
+void LimitedNodeList::updateLocalSocket() {
+    // when update is called, if the local socket is empty then start with the guessed local socket
+    if (_localSockAddr.isNull()) {
+        setLocalSocket(HifiSockAddr { getGuessedLocalAddress(), _nodeSocket.localPort() });
+    }
+
+    // attempt to use Google's DNS to confirm that local IP
+    static const QHostAddress RELIABLE_LOCAL_IP_CHECK_HOST = QHostAddress { "8.8.8.8" };
```
Comment (sethalves) [inline libraries/networking/src/LimitedNodeList.cpp]: This makes me nervous, but I guess we'll see how it goes.

**Pull request merged**